app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/RerouteActivity.java

```diff
@@ -214,6 +215,15 @@ public void userOffRoute(Location location) {
 
   @Override
   public void onProgressChange(Location location, RouteProgress routeProgress) {
+    boolean inTunnel = routeProgress.inTunnel();
```

Comment:
This chunk of logic can be extracted into a `private` method.  Maybe we can think of better naming for `isInTunnel` vs. `inTunnel` - it's a little confusing to determine which variable is coming from where.